src/Components/test/E2ETest/ServerExecutionTests/RemoteRendererBufferLimitTest.cs

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
Wow, these are very detailed tests!

Do you think it would be practical for any of the coverage to be in unit tests in `RemoteRendererTest`? I would have *guessed* that would have been easier to avoid things like flakiness with timings, but maybe it's impractical for some reason.

Reply:
I started with the E2E tests and after that I didn't feel it was worth adding unit tests too. I mostly did E2E tests (although they could have been unit tests) because I wanted to make sure the actual behavior matched my assumptions. Specially with regards to the view output, etc. and having E2E tests where I can simply fire the app and see the browser helped me do that. Then I simply put some tests around it and called it good.

I'm not super concerned about the timings because the assertions I make are broad on purpose (I don't check for exact values but for trends) and I feel the tests are more understandable this way.